File: OOP/python/encapsulation.py
class Encapsulation:
    def __init__(self, name, age):
        self.__name = name
        self.__age = age

    def getName(self):
        return self.__name

    def getAge(self):
        return self.__age

    def setName(self, name):
        self.__name = name

    def setAge(self, age):
        self.__age = age

    def __str__(self):
        return "Name: " + self.__name + " Age: " + str(self.__age)

if __name__ == "__main__":
    print("App Started")
    # encapsulation = Encapsulation("John", 30)
    # print(encapsulation)
    # encapsulation.setName("Jane")
    # encapsulation.setAge(31)
    # # encapsulation.__age(32)
    # print(encapsulation)
    print("App Ended")

File: OOP/c++/encapsulation.cpp
#include <string.h>

#include <iostream>

class Encapsulation {
   private:
    std::string name;
    int age;

   public:
    Encapsulation() {
        this->name = "";
        this->age = 0;
    }

    void setName(std::string name) {
        this->name = name;
    }
    void setAge(int age) {
        this->age = age;
    }

    std::string getName() {
        return name;
    }

    int getAge() {
        return age;
    }

    void print() {
        std::cout << "Name: " << name << " Age: " << age << std::endl;
    }
};

int main() {
    std::cout << "App Started" << std::endl;
    Encapsulation app;
    app.print();
    app.setName("John");
    app.setAge(30);
    app.print();
    // app.age = 5; // // error because age is private
    std::cout << "App Ended" << std::endl;
    return 0;
}